packages/data-stores/src/site/actions.ts

```diff
@@ -127,15 +127,19 @@ export function createActions( clientCreds: WpcomClientCredentials ) {
 
 	function* setSiteTitle( siteId: number, title: string ) {
 		try {
+			// update the UI optimistically
+			yield receiveSiteTitle( siteId, title );
```

Comment:
This doesn't feel right here. Let's leave the optimistic UI update to be handled at the component level